omnibus/config/software/datadog-agent-integrations-py3.rb

```diff
@@ -172,6 +172,27 @@
     end
 
     checks_to_install.each do |check|
+      # pip < 21.2 replace underscores by dashes in package names per https://pip.pypa.io/en/stable/news/#v21-2
+      # whether or not this might switch back in the future is not guaranteed, so we check for both name
+      # with dashes and underscores
+      if !(installed_list.include?(check) || installed_list.include?(check.gsub('_', '-')))
+        if windows_target?
+          shellout! "#{python} -m pip wheel . --no-deps --no-index --wheel-dir=#{wheel_build_dir}", :env => build_env, :cwd => "#{windows_safe_path(project_dir)}\\#{check}"
+        else
+          shellout! "#{python} -m pip wheel . --no-deps --no-index --wheel-dir=#{wheel_build_dir}", :env => build_env, :cwd => "#{project_dir}/#{check}"
+        end
+        shellout! "#{python} -m pip install datadog-#{check} --no-deps --no-index --find-links=#{wheel_build_dir}"
+        if cache_bucket != '' && ENV.fetch('INTEGRATION_WHEELS_SKIP_CACHE_UPLOAD', '') == '' && cache_branch != nil
+          shellout! "inv -e agent.upload-integration-to-cache " \
+                    "--python 3 --bucket #{cache_bucket} " \
+                    "--branch #{cache_branch} " \
+                    "--integrations-dir #{windows_safe_path(project_dir)} " \
+                    "--build-dir #{wheel_build_dir} " \
+                    "--integration #{check}",
+                    :cwd => tasks_dir_in
+        end
+      end
```

Comment:
To reviewers: we need to do that first in case the integration is not retrieved from the cache. Then, we can copy the files and drop them if needed